code/data_structures/src/tree/binary_tree/avl_tree/avl__tree.py

```diff
@@ -0,0 +1,262 @@
+#import random, math
+
+outputdebug = False 
+
+def debug(msg):
+    if outputdebug:
+        print msg
+
+class Node():
+    def __init__(self, key):
+        self.key = key
+        self.left = None 
+        self.right = None 
+
+
+
+
+class AVLTree():
+    def __init__(self, *args):
+        self.node = None 
+        self.height = -1  
+        self.balance = 0; 
+        
+        if len(args) == 1: 
+            for i in args[0]: 
+                self.insert(i)
+                
+    def height(self):
+        if self.node: 
+            return self.node.height 
+        else: 
+            return 0 
+    
+    def is_leaf(self):
+        return (self.height == 0) 
+    
+    def insert(self, key):
+        tree = self.node
+        
+        newnode = Node(key)
+        
+        if tree == None:
+            self.node = newnode 
+            self.node.left = AVLTree() 
+            self.node.right = AVLTree()
+            debug("Inserted key [" + str(key) + "]")
```

Comment:
Please remove all debug statements as they aren't exactly part of the final program.

Reply:
All debug statements from code removed